### PR TITLE
Add additional tests for TensorDataset

### PR DESCRIPTION
**Pull request by stes**
PRs #166 and #180 did some changes to the TensorDataset class without adding tests. This PR adds such tests to cover the basic functionality (type checking, init of the dataset, accessing the data and reproducing the issue described in #174 ).

This brought two minor errors to light which are also fixed in the PR (typo in `self._to_tensor(discrete, check_dtype="integer")`, wrong docstring, exception handling.

### CODE REVIEW

**Comment by stes** [comment]
Ok, funny edge case, good we are running windows tests
image

int32 vs int64 defaults are handled differently on windows, will add that explicitly then.

File: cebra/data/datasets.py [hunk 1]
```diff
@@ -98,11 +98,16 @@ def _to_tensor(
...
+            if check_dtype not in ["int", "float"]:
+                raise ValueError(
+                    f"check_dtype must be 'int' or 'float', got {check_dtype}")
             if (check_dtype == "int" and not cebra_helper._is_integer(array)
                ) or (check_dtype == "float" and
                      not cebra_helper._is_floating(array)):
                 raise TypeError(
                     f"Array has type {array.dtype} instead of {check_dtype}.")
+        if cebra_helper._is_floating(array):
+            array = array.float()
```
Comment (stes) [inline cebra/data/datasets.py:110]:
```suggestion
        if cebra_helper._is_floating(array):
            array = array.float()
        if cebra_helper._is_integer(array):
            # NOTE(stes): Required for standardizing number format on
            # windows machines.
            array = array.long()
```
